src/components/ComputeModal.js

```diff
@@ -951,22 +970,22 @@ export const ComputeModalBase = ({ onDismiss, onSuccess, runtimes, persistentDis
                   h(Select, {
                     id,
                     isSearchable: false,
-                    value: sparkMode,
+                    value: runtimeType,
                     onChange: ({ value }) => {
-                      setSparkMode(value)
-                      updateComputeConfig('componentGatewayEnabled', !!value)
+                      setRuntimeType(value)
+                      updateComputeConfig('componentGatewayEnabled', isDataproc(value))
                     },
                     options: [
-                      { value: false, label: 'Standard VM', isDisabled: requiresSpark },
-                      { value: 'master', label: 'Spark master node', isDisabled: requiresGCE() },
-                      { value: 'cluster', label: 'Spark cluster', isDisabled: requiresGCE() }
+                      { value: runtimeTypes.gceVm, label: runtimeTypes.gceVm, isDisabled: requiresSpark },
```

Comment:
Nit: I believe you don't need to specify label if it is the same as value.

Reply:
Thanks!